doc/contributing/documentation_guide.md

```diff
@@ -503,3 +513,66 @@ mention `Hash#fetch` as a related method, and `Hash#merge` might mention
 For methods that accept multiple argument types, in some cases it can
 be useful to document the different argument types separately.  It's
 best to use a separate paragraph for each case you are discussing.
+
+## Further Considerations
+
+### Case Treatments
+
+When a series of paragraphs documents _cases_ that form a _partition_,
+consider using bullets (unordered items) to emphasize that structure;
+such structures may be nested.
+
+The documentation for Array.new is organized thus (some text omitted)
+
+```
+Returns a new +Array+ object:
+
+- With no block given:
+
+    - With no argument given, returns ...:
+    - With argument +array+ given, returns ...:
+    - With numeric argument +size+ given, returns ...:
+
+- With a block given, returns ...:
+
+Raises TypeError if ....
+```
+
+### Trivial \Method Behaviors
+
+For a trivial method behavior,
+it's usually best not to include that behavior in the discussion;
+consider whether it is useful and convenient to show that behavior
+as part of the example code.
+
+In Array.new, the examples for argument `size` include this trivial result:
+
+```
+Array.new(0)    # => []
+```
+
+### Exceptions
+
+Documentation may include discussions of exceptions that may be raised.
+Consider whether it is useful and convenient to include an exceptional behavior in the examples
+(instead of in the discussion).
+
+In Array.new, the examples for argument `size` include this exceptional behavior:
+
+```
+Array.new(-1)   # Raises ArgumentError (negative array size).
+```
```

Comment:
In the case were the exception is not obvious, I think it's better to explain instead of just giving an example.  In the Array.new example, maybe it isn't needed, since when you see the argument and message, you can understand why, but that may not be true in more complex cases (and cases where the exception isn't obvious are often complex).